merge.py

```diff
@@ -61,129 +62,146 @@ def merge(self):
-        repo_url = GITHUB_REPO_URL.format(GITHUB_ORG, repo)
-        clone_args = ['git', 'clone', '-b', TARGET_BRANCH, repo_url, tmp_dir]
-        clone_success = call(clone_args, cwd=tmp_dir)
-        if clone_success != 0:
-            pr.post_error(
-                'Couldn\'t clone from github/{}/{}. Please try again.'.format(
-                    GITHUB_ORG, repo))
+        self.tmp_dir = TMP_DIR_FMT.format(dir='{}-{}'.format(
+            repo, pr.get_num()))
+        try:
+            _set_up(self.tmp_dir)
+        except AssertionError:
+            pr.post_error('Setup of initial directory failed, try again.')
             return False
-
-        call(['git', 'remote', 'add', TARGET_REMOTE, APACHE_URL.format(repo)],
-             cwd=tmp_dir)
-        call(['git', 'remote', 'rename', 'origin', SOURCE_REMOTE], cwd=tmp_dir)
-        call('git config --local --add remote.' + SOURCE_REMOTE +
-             '.fetch "+refs/pull/*/head:refs/remotes/{}/pr/*"'.format(
-                 SOURCE_REMOTE), shell=True,
-             cwd=tmp_dir)
-        call(['git', 'fetch', '--all'], cwd=tmp_dir)
-        _print_flush('Initial work complete.')
-
-        # Clean up fetch
-        pr_name = 'finish-pr-{}'.format(pr)
-        ic_args = ['git', 'checkout', '-b', pr_name, 'github/pr/{}'.format(pr)]
-        initial_checkout = call(ic_args, cwd=tmp_dir)
-        if initial_checkout != 0:
-            pr.post_error("Couldn't checkout code. Please try again.")
+        try:
+            repo_url = GITHUB_REPO_URL.format(org='apache', repo=repo)
+            self.run('git clone -b {branch} {repo}'.format(
+                branch=self.config['merge_branch'], repo=repo_url), 'Clone',
+                     'Clone failed. Please try again.')
+            self.run('git remote add apache {apache_url}'.format(
+                apache_url=APACHE_GIT.format(repo=repo)), 'Add Remote',
+                     'Failed to add remote. Please try again.')
+            self.run('git remote rename origin github', 'Rename Origin',
+                     'Failed to rename origin. Please try again.')
+            self.run('git config --local --add remote.github.fetch '
+                     '"+refs/pull/*/head:refs/remotes/github/pr/*"',
+                     'Configure git fetch.', 'Failed to configure git fetch. '
+                                             'Please try again.', shell=True)
+            self.run('git fetch --all', 'Fetch everything.',
+                     'Fetch failed. Please try again.')
+            pr_num = pr.get_num()
+            pr_name = 'finish-pr-{}'.format(pr.get_num())
+            self.run('git checkout -b {pr_name} github/pr/{pr_num}'.format(
+                pr_name=pr_name, pr_num=pr_num),
+                     'Checkout PR', 'Failed to check out PR. Please try again.')
+            branch = '{}/{}'.format('apache', self.config['merge_branch'])
+            self.run('git rebase {branch}'.format(branch=branch),
```

Comment:
Looking better (easier to see the sequence of commands). I'd propose some readability tweaks:

1. Use keyword arguments for everything except the command
2. Consider taking all the variables that are likely to be useful in messages/commands (eg., branch) and putting them in a single map, that self.run automatically passes to format. This way you don't need to keep modifying that as the things being used change.

```
self.run('git rebase {branch}',
    desc='Rebase against target branch.',
    failureMessage='Automatic rebase failed. Please manually rebase '
    'against {branch} and try again.')
```

If you don't want the environment to be shared across all the run commands, at least consider passing it to run and allowing it to format all the strings.